topics/go/language/embedding/exercises/exercise1/exercise1.go

```diff
@@ -58,8 +58,7 @@ func NewCachingFeed(f *Feed) *CachingFeed {
 
 // Fetch calls the embedded type's Fetch method if the key is not cached.
 func (r *CachingFeed) Fetch(key string) (Document, error) {
-	doc, ok := r.docs[key]
-	if ok {
+	if doc, ok := r.docs[key]; ok {
```

Comment:
Can we use the idiom to use the negative path in the if statement. I have to look closer to the function but it just occurred to me we are not following this.

Reply:
I've tried it and it doesn't seem any better to me.

We *are* being idiomatic here. Don't focus on "negative path" for the idiom. To me the idiom is more about returning early which is what we're doing here. Typically those early returns come in the form of handling errors but it is also very common to see optimizations return early as well.

```
if it's in the cache {
   return early
}

normal cache-miss implementation
```

See https://golang.org/src/strings/strings.go?s=2020:2171#L68 for a stdlib example

Reply:
I agree and it's why I said can we use it. I should have looked at the code first myself.